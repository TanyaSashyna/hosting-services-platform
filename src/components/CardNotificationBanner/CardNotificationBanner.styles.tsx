import styled from "styled-components";
import { paddings } from "@project-name-frontend/ui-package";

export const Container = styled("div")`
  padding: ${paddings.default};
  padding-top: 0;
`;
