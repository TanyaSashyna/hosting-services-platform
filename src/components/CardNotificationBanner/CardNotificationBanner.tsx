import React, { FC } from "react";
import { NotificationBanner, Link } from "@project-name-frontend/ui-package";

import { Container } from "./CardNotificationBanner.styles";

type CardNotificationBannerProps = {
  hostReachable: boolean;
  status: string;
  canChangeDns: boolean;
};

export const CardNotificationBanner: FC<CardNotificationBannerProps> = ({
  hostReachable,
  status,
  canChangeDns,
}) => {
  if (!hostReachable) {
    return (
      <Container>
        <NotificationBanner
          icon="warning"
          intent="warning"
          border={true}
          title="Warning"
          body={
            <>
              Unable to determine origin web server/web server IP. Please
              contact the Technical Support team at{" "}
              <Link href="mailto:support@project-name.com">support@project-name.com</Link>{" "}
              for assistance.
            </>
          }
        />
      </Container>
    );
  }

  if (!canChangeDns) {
    return (
      <Container>
        <NotificationBanner
          icon="warning"
          intent="warning"
          border={true}
          title="Warning"
          body={
            status === "enabled" ? (
              <>
                Your DNS records arenâ€™t hosted by project-name. We highly recommend
                updating these to their pre-Tool configuration before
                cancellation.
              </>
            ) : (
              <>
                Once Tool setup is complete, please follow the instructions to
                update your DNS records and fully activate your service.
              </>
            )
          }
        />
      </Container>
    );
  } else {
    return (
      <Container>
        <NotificationBanner
          icon="info"
          intent="neutral"
          border={true}
          title="Note"
          body={
            status === "enabled"
              ? "Cancelling Tool automatically reverts your DNS records to their pre-Tool configuration."
              : "Adding Tool to your domain automatically updates your DNS records to point to Tool."
          }
        />
      </Container>
    );
  }
};

CardNotificationBanner.displayName = "CardNotificationBanner";
