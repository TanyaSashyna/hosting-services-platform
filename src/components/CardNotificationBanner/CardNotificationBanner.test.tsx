import React from "react";
import { cleanup, render, waitFor } from "@testing-library/react";

import { CardNotificationBanner } from "./CardNotificationBanner";

afterEach(cleanup);

describe("CardNotificationBanner", () => {
  it("renders CardNotificationBanner and matches snapshot", () => {
    const { container, getByText, asFragment } = render(
      <CardNotificationBanner
        hostReachable={false}
        canChangeDns={true}
        status="enabled"
      />
    );

    expect(container).toBeDefined();
    expect(
      waitFor(() =>
        expect(
          getByText(
            "Cancelling Tool automatically reverts your DNS records to their pre-Tool configuration."
          )
        )
      )
    ).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders when status prop is disabled", () => {
    const { getByText } = render(
      <CardNotificationBanner
        hostReachable={true}
        canChangeDns={true}
        status="disabled"
      />
    );

    expect(
      waitFor(() =>
        expect(
          getByText(
            "Adding Tool to your domain automatically updates your DNS records to point to Tool."
          )
        )
      )
    ).toBeDefined();
  });

  it("renders when status is enabled and canChangeDns props is true", () => {
    const { getByText } = render(
      <CardNotificationBanner
        hostReachable={true}
        canChangeDns={false}
        status="enabled"
      />
    );

    expect(
      waitFor(() =>
        expect(
          getByText(
            "Your DNS records arenâ€™t hosted by project-name. We highly recommend updating these to their pre-Tool configuration before cancellation."
          )
        )
      )
    ).toBeDefined();
  });

  it("renders when status is disabled and canChangeDns is true", () => {
    const { getByText } = render(
      <CardNotificationBanner
        hostReachable={true}
        canChangeDns={false}
        status="disabled"
      />
    );

    expect(
      waitFor(() =>
        expect(
          getByText(
            "Once Tool setup is complete, please follow the instructions to update your DNS records and fully activate your service."
          )
        )
      )
    ).toBeDefined();
  });

  it("renders when hostReachable prop is false", () => {
    const { getByText } = render(
      <CardNotificationBanner
        hostReachable={false}
        canChangeDns={true}
        status="disabled"
      />
    );

    expect(getByText("support@project-name.com")).toBeDefined();
  });
});
