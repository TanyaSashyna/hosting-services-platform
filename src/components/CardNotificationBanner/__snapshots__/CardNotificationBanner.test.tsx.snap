// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardNotificationBanner renders CardNotificationBanner and matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  padding: 16px;
  padding-top: 0;
}

<div
    class="c0"
  >
    <div
      class="notificationBanner notificationBanner__border--warning notificationBanner--borderAll"
    >
      <div
        class="notificationBanner__icon"
      >
        <i
          class="icon icon--warning"
        >
          warning
        </i>
      </div>
      <div
        class="notificationBanner__content"
      >
        <div
          class="notificationBanner__content__title"
        >
          Warning
        </div>
        <div
          class="unotificationBanner__content__body"
        >
          Unable to determine origin web server/web server IP. Please contact the Technical Support team at 
          <a
            class="link link--type-default"
            href="mailto:support@project-name.com"
          >
            support@project-name.com
          </a>
           for assistance.
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
